ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import {
+  MatExpansionPanel,
+  MatTabChangeEvent
+} from '@angular/material'
+
+import {JobMetadataResponse} from '../../shared/model/JobMetadataResponse';
+import {ResourceUtils} from '../../shared/utils/resource-utils';
+import {GcsService} from '../../core/gcs.service';
+
+@Component({
+  selector: 'jm-resources',
+  templateUrl: './resources.component.html',
+  styleUrls: ['./resources.component.css'],
+})
+export class JobResourcesComponent implements OnChanges {
+  @Input() job: JobMetadataResponse;
+
+  script: string = '';
+  inputs: Array<string> = [];
+  logFileData: Map<string, string> = new Map();
+  logFiles: Array<string> = [];
+  outputs: Array<string> = [];
+  currentResourcesTab: string = '';
+
+  constructor(private readonly gcsService: GcsService) {}
+
+  ngOnChanges(changes: SimpleChanges) {
```

Comment:
Why `onChanges` (vs `onInit`) here? Is the job metadata response not necessarily present at initialization time?

Reply:
Ah, this was likely a carry-over from the pre-Resolver days where the job would late-load in.